### ISSUE TITLE
Feature request: Add build in cloud service to store and access templates and scripts

### ISSUE THREAD

**Issue by multiwebinc** [thundernest/quicktext#54]
I work at home and at work and in both places I need my templates files, and I need changes that are done in one location to be applied to the other location. Currently I use [Seafile](https://github.com/haiwen/seafile) to sync that directory, but if there were a built-in way to accomplish this, that would be better, even if it needs to be a premium service that charges a monthly fee to pay for cloud storage, it would probably be worth it for many users. Also, watching for changes in the cloud and updating the local files in real time without the need to restart Thunderbird would also be a big plus.

**Comment by jobisoft** [comment]
The "Import on startup" feature, could do the job. Just place your templates on some server and import them on each start of TB. You would need to manually update the templates on the server, but all your "clients" will use the same templates.

**Comment by multiwebinc** [comment]
@jobisoft Yes, I saw that feature as well, but as you said, it would require manually editing the templates, which is not convenient at all.

**Comment by jobisoft** [comment]
Why is your seafile approach not working? You could set up two groups, one with wri

**Comment by jobisoft** [comment]
Hit the wrong button!

**Comment by jobisoft** [comment]
Why is your seafile approach not working? Seafile has a permission management, so only one system gets write permissions, all the others can only read the template files and import that on start?

**Comment by multiwebinc** [comment]
@jobisoft Seafile is working fine for me, but not everybody has the resources or ability to set up a Seafile server somewhere. But signing up for a Quicktext sync service integrated into the plugin as a premium service for a few dollars/month could generate extra revenue.

**Comment by jobisoft** [comment]
Ah ok, so you do not need any help, it is working for you and this is a feature request to simplify it for others.

**Comment by multiwebinc** [comment]
> Ah ok, so you do not need any help, it is working for you and this is a feature request to simplify it for others.

Yes, exactly.

**Comment by Fryday80** [comment]
I think something like this would be great. We use it in a 3-pc-network at work.  For my usage it would be enough if you could specify a directory (outside TB) where your templates are stored. The export / import function isn't satisfying because last time I did it this way the result was puzzling... just half of the templates appeared and there were mistakes in them.
Perhaps this change would fix multiwebinc's problem as well and the change to choose a special folder might be implemented easy.

**Comment by jobisoft** [comment]
When was the last time you tried? We made changes to load/save methods, maybe that fixed your issues?

You can specify a local folder there or a mounted drive on some other server (mounted into the local filesystem).

**Comment by argv-minus-one** [comment]
What about storing Quicktext templates as emails in a folder on the IMAP server? That's what Thunderbird does with its templates.

**Comment by jakubstaszak-com** [comment]
Hi, using built-in cloud per se is not a good idea.
First and foremost that is huge cost of implementation.
Second, not everyone wants to share their data with any random cloud.
Third, using currently available import/export method couldn't be easier. Onedrive, Google drive, Dropbox.... pick one and share as any other resource.
Also, it's not exactly like a salesman is going to deliberately or accidentally burst into all the scripts and modify them all, is it?
Last but not least, even if, there is a restore function available in most of those cloud solutions, isn't it?

**Comment by jobisoft** [comment]
@jakubstaszak-com : Well, nobody said it is mandetory to use cloud services, but I do see the need to allow it optional. Either with any self hosted WebDAV server or a partner service if the user does not want to manage his own server. Actually working on that right now.

**Comment by userQAY** [comment]
i use google drive to sync the templates and scripts, but, it is annoying, need to mannualy export (twice, one for templates and one for scripts) every change, i made. I would appreciate: 
1/ automatic export every one hour, day , etc. ,  (sync via google drive)
2/ 2-way sync, so i can make changes on both (all) PCs. (optional).  With simply prompt "templates has been changed on other PC, do you want import? " every I start TB. (or TB can check date of file every hour(etc))

**Comment by AlbertronicDev** [comment]
Syncing is very welcome, especially when working with multiple users sharing, adding and and updating the same templates. However, cloud services is not the best option as this may increase monthly costs for the user, will need increased developments to keep the cloud service(s) supported in Quicktext etc.
@argv-minus-one suggestion to use IMAP to store the templates is a great idea. Local storage, quick syncing etc. The mailserver essentially is your cloud server :)

**Comment by jobisoft** [comment]
That is interesting. I recently was involved in updating [iOS IMAP Notes](https://addons.thunderbird.net/de/thunderbird/addon/ios-imap-notes/?src=search), which shows that this concept does work. I will have a deeper look.

**Comment by jobisoft** [comment]
I cannot believe this request is already 7 years old.

With the new provider based storage approach of the next quicktext, it is now a user choice, where the templates and associated files should be stored. They can still keep them local to the extension, on the local home folder, or in the cloud. We will get different cloud storage providers, like Nextcloud, Seafile, and what not. Someone could even implement an IMAP storage provider.

The missing piece is a provider tailored to users who do not want to fiddle with setting up their own cloud servers or do not want to deal with a complicated setup process. I therefore decided to launch my own cloud service. Setup and sign-up happens directly in Quicktext with just a few mouse clicks. The files are stored securely & encrypted in Europe and users can access their templates from different Thunderbird installations. It is reduced to the minimum, no sharing of files outside of Thunderbird, no web interface. 

It is of course not free. In the current calculation, the smallest plan will cover 4GB and will cost about 1,49EUR per month. 

Is there still a demand for such a service? It is possible and the work needed to get it running is about 75% done. I have a proof-of-concept implementation running.

**Comment by jobisoft** [comment]
With v6.6 released, you can now use any WebDAV server to store your templates. There are also some free variants, for example GMX has 4GB in their free plans. I therefore think that an additional Quicktext branded service is not needed.
